mm2src/coins/utxo.rs

```diff
@@ -1294,11 +1411,91 @@ impl Default for ElectrumBuilderArgs {
     }
 }
 
+#[derive(Debug)]
+pub enum PrivKeyPolicy {
```

Comment:
It seems to me that `PrivKeyPolicy` should not be utxo specific. We will likely use the same approach for all supported protocols so I propose making it generic over keypair type as follows:
```rust
pub enum PrivKeyPolicy<T> {
    KeyPair(T),
    HardwareWallet,
}
```

and reuse in all coins, which support hardware wallets. Using this approach, we can avoid duplication in methods like `display_privkey` as follows:
```rust
trait CoinWithPrivkeyPolicy {
    type KeyPair;

    fn get_priv_key_policy(&self) -> PrivKeyPolicy<Self::KeyPair>;

    fn display_priv_key(&self, key_pair: &Self::KeyPair) -> String;
}

trait DisplayPrivKeyRpc {
    fn display_privkey_rpc(&self) -> Result<...>
}

impl<T: CoinWithPrivkeyPolicy> DisplayPrivKey for T {
    fn display_privkey_rpc(&self) -> Result<..> {
         match self.get_priv_key_policy {
             PrivKeyPolicy::KeyPair(key_pair) => Ok(self.display_priv_key(key_pair)),
             PrivKeyPolicy::HardwareWallet => Err("Coin is initialized using HW wallet"),
         }
    }
}
```

What do you think? :slightly_smiling_face:

Reply:
You are absolutely right!

Reply:
It will take some time to implement the `CoinWithPrivkeyPolicy` trait for all coins since first we should replace `my_address` and `priv_key` fields with `priv_key_policy` for ETH/QRC20 coins.
As of now, I moved `PrivKeyPolicy` and `DerivationPath` to `lp_coins`.

Reply:
Ok, that's fine for now. Please do this refactoring on the next iteration. I will keep this unresolved.